core/src/main/java/me/mykindos/betterpvp/core/combat/listeners/CombatListener.java

```diff
@@ -320,6 +326,16 @@ public void startCustomDamageEvent(EntityDamageEvent event) {
 
         LivingEntity damager = getDamagerEntity(event);
         Entity damaging = getDamagingEntity(event);
+        if (event.getCause() == DamageCause.SUFFOCATION) {
+            Optional<RestoreBlock> restoreBlockOptional = worldBlockHandler.getRestoreBlock(damagee.getEyeLocation().getBlock());
+            if (restoreBlockOptional.isPresent()) {
+                Player newDamager = restoreBlockOptional.get().getSummoner();
+                if (newDamager != null) {
+                    damager = newDamager;
+                }
+            }
+        }
+
```

Comment:
Instead of doing this, listen to customdamageevent in the RestoreBlockListener and set the damager that way (if damager IS null)

Reply:
implemented